src/smc-util/misc.coffee

```diff
@@ -1898,3 +1898,17 @@ exports.bind_objects = (scope, arr_objects) ->
                 return bound_func
             else
                 return val
+
+# Provide the index that a value would be inserted
+# into an array where the val is between the number before
+# and after it
+exports.array_bisect = (arr, val) ->
```

Comment:
Also this doesn't seem to be doing any bisecting. It's more like the index finder for an insertion sort. Maybe name it `find_larger_entry`. This naming would be analogous to [Array.prototype.find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find). Which maybe this should use?